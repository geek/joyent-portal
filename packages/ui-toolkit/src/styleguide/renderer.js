import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import remcalc from 'remcalc';
import is from 'styled-is';

import { Grid, Row, Col } from '../grid';
import theme from '../theme';

const Main = styled(Row)`
  padding-top: ${remcalc(40)};
`;

const Sticky = styled.div`
  position: sticky;
  top: 0;
  z-index: 9;

  ${is('sidebar')`
    top: ${remcalc(100)};
    max-height: 85vh;
    overflow: auto;
  `};
`;

const fullTheme = {
  ...theme,
  spacing: {
    0.5: remcalc(4),
    0: remcalc(0),
    1: remcalc(6),
    2: remcalc(12),
    3: remcalc(18),
    4: remcalc(24),
    5: remcalc(30),
    6: remcalc(36),
    7: remcalc(42),
    8: remcalc(48),
    9: remcalc(54),
    10: remcalc(60)
  }
};

const StyleGuideRenderer = ({
  title,
  homepageUrl,
  children,
  toc,
  hasSidebar
}) => (
  <ThemeProvider theme={fullTheme}>
    <Grid>
      <Main>
        {hasSidebar && (
          <Col xs={3}>
            <Sticky sidebar>{toc}</Sticky>
          </Col>
        )}
        <Col xs={hasSidebar ? 9 : 12} lg={hasSidebar ? 8 : 12}>
          {children}
        </Col>
      </Main>
    </Grid>
  </ThemeProvider>
);

export default StyleGuideRenderer;
