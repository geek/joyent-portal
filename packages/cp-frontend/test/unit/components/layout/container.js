/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Container from '@components/layout/container';

it('renders <Container /> without throwing', () => {
  const tree = renderer
    .create(
      <Container />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
