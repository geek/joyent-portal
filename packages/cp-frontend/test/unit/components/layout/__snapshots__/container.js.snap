// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Container /> without throwing 1`] = `
.c0 {
  margin-right: auto;
  margin-left: auto;
  padding-top: 1.1875rem;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: block;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

@media only screen and (min-width:48em) {
  .c0 {
    width: 46rem;
  }
}

@media only screen and (min-width:64em) {
  .c0 {
    width: 61rem;
  }
}

@media only screen and (min-width:75em) {
  .c0 {
    width: 76rem;
  }
}

<div
  className="c0"
/>
`;
